File: ts-sdk/src/sdk/models/operations/getfilesidmetadataidid.ts
import { SpeakeasyMetadata, SpeakeasyBase, RetryConfig } from "../../../internal/utils";
import * as shared from "../shared";


export enum GetFilesIdMetadataIdIdScopeEnum {
    Global = "global",
    Enterprise = "enterprise"
}


export class GetFilesIdMetadataIdIdPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=file_id" })
  fileId: string;

  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=scope" })
  scope: GetFilesIdMetadataIdIdScopeEnum;

  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=template_key" })
  templateKey: string;
}


export class GetFilesIdMetadataIdIdRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  retries?: RetryConfig;

  @SpeakeasyMetadata()
  pathParams: GetFilesIdMetadataIdIdPathParams;
}


export class GetFilesIdMetadataIdIdResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  clientError?: shared.ClientError;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  metadata?: shared.Metadata;

  @SpeakeasyMetadata()
  statusCode: number;
}

File: go-sdk/pkg/models/operations/getfilesidmetadataidid.go
package operations

import (
	"openapi/pkg/models/shared"
	"openapi/pkg/utils"
)

type GetFilesIDMetadataIDIDScopeEnum string

const (
	GetFilesIDMetadataIDIDScopeEnumGlobal     GetFilesIDMetadataIDIDScopeEnum = "global"
	GetFilesIDMetadataIDIDScopeEnumEnterprise GetFilesIDMetadataIDIDScopeEnum = "enterprise"
)

type GetFilesIDMetadataIDIDPathParams struct {
	FileID      string                          `pathParam:"style=simple,explode=false,name=file_id"`
	Scope       GetFilesIDMetadataIDIDScopeEnum `pathParam:"style=simple,explode=false,name=scope"`
	TemplateKey string                          `pathParam:"style=simple,explode=false,name=template_key"`
}

type GetFilesIDMetadataIDIDRequest struct {
	Retries    *utils.RetryConfig
	PathParams GetFilesIDMetadataIDIDPathParams
}

type GetFilesIDMetadataIDIDResponse struct {
	ClientError *shared.ClientError
	ContentType string
	Metadata    *shared.Metadata
	StatusCode  int64
}
